libvuln/updates/manager.go

```
@@ -193,6 +196,16 @@ func (m *Manager) Run(ctx context.Context) error {
 	// all in-flight go routines are gauranteed to release their sems.
 	sem.Acquire(context.Background(), int64(m.batchSize))
 
+	// checking >= 2 is a defensive tactic, we want to always keep min 2 update operations
+	// for notification purposes.
+	if m.updateRetention >= 2 {
```

The constructor should return an error if this `== 1`.